# HG changeset patch
# User Rik <rik@octave.org>
# Date 1544491324 28800
#      Mon Dec 10 17:22:04 2018 -0800
# Node ID d57d30caffd608b6110032f13213cf2dc134ee61
# Parent  95633ec174cf6574d884c3b9a3a08e9a676d83b0
Add BISTS tests for iscomplex().

* data.cc (Fiscomplex): Add BIST tests.

diff --git a/libinterp/corefcn/data.cc b/libinterp/corefcn/data.cc
--- a/libinterp/corefcn/data.cc
+++ b/libinterp/corefcn/data.cc
@@ -3246,16 +3246,35 @@ Return true if @var{x} is a complex-valu
 @end deftypefn */)
 {
   if (args.length () != 1)
     print_usage ();
 
   return ovl (args(0).iscomplex ());
 }
 
+
+/*
+%!assert (iscomplex (4), false)
+%!assert (iscomplex (i), true)
+%!assert (iscomplex (4+3i), true)
+%!assert (iscomplex ([1, 2, 3]), false)
+%!assert (iscomplex ([1, 2i, 3]), true)
+
+%!assert (iscomplex (0j), false)
+%!assert (iscomplex (complex (0,0)), true)
+%!assert (iscomplex ("4"), false)
+%!assert (iscomplex ({i}), false)
+
+## Test input validation
+%!error iscomplex ()
+%!error iscomplex (1, 2)
+*/
+
+
 DEFUN (isfloat, args, ,
        doc: /* -*- texinfo -*-
 @deftypefn {} {} isfloat (@var{x})
 Return true if @var{x} is a floating-point numeric object.
 
 Objects of class double or single are floating-point objects.
 @seealso{isinteger, ischar, islogical, isnumeric, isstring, isa}
 @end deftypefn */)
